### ISSUE TITLE
Console warning: prop amPm is required but not set in Hour12Input

### ISSUE THREAD

**Issue by emilyuhde** [wojtekmaj/react-time-picker#24]
When using the TimePicker in dev mode, I'm seeing a console warning.

`Warning: Failed prop type: The prop amPm is marked as required in Hour12Input, but its value is null.
    in Hour12Input (created by TimeInput)
    in TimeInput (created by TimePicker)
    in div (created by TimePicker)`

I only seem to be seeing this warning when I use hooks to manage the state of the TimePicker component: https://gist.github.com/emilyuhde/ded5df5a5b54baeb981403148b9c892a

I don't see the warning when I use a stateful class: https://gist.github.com/emilyuhde/8275a185ca2ca649a30dc3fb6df9be88

**Comment by emilyuhde** [comment]
I think I see the issue, the react dependency is  `"react": "^16.7.0"`. Are there plans to update the dependency to 16.8.x soon?

**Comment by wojtekmaj** [comment]
`^16.7.0` signifies compatibility with `>=16.7.0`; `<17.0.0`, so with `16.8.0`, too. I don't think that is the cause!

**Comment by wojtekmaj** [comment]
Hmmmm, I tested your code and I don't see the warning you're seeing. The only thing I see is that you use `useState()` without default prop, which causes it to be `undefined` initially, which is not the case in your class component. So maybe that's the problem?

**Comment by emilyuhde** [comment]
Thanks for looking at this. I was looking at this issue some more yesterday and I think I figured out what's actually causing it. If I initialize the `value` to null, and the TimePicker is in 12 hour format, that seems to be what's causing the console warning to appear. I think the use of a class component with state vs using hooks was just a red herring and isn't actually causing an issue. That's why I closed this issue and created #31 since that console warning isn't wrong, it's just annoying when I want to intentionally initialize the value to null so that the TimePicker will display the placeholder --:-- -- state.
